tests/tools/fio/fio.go

```diff
@@ -54,27 +139,82 @@ func NewRunner() (*Runner, error) {
 					"blocksize":         "1m",
 					"refill_buffers":    "",
 					"rw":                "write",
-					"unique_filename":   "1",
-					"directory":         dataDir,
-				},
+				}.WithDirectory(fioWriteBaseDir),
 			},
 		},
-	}, nil
+	}
+
+	err = fr.verifySetupWithTestWrites()
+	if err != nil {
+		log.Printf("Verify environment setup:\n")
+		log.Printf("   Set %s (=%q)to the fio executable\n", FioExeEnvKey, exeStr)
+		log.Printf("   - OR -\n")
+		log.Printf("   Set %s (=%q) to the fio docker image", FioDockerImageEnvKey, imgStr)
+		log.Printf("   Set %s (=%q) to the path where fio data will be used locally", LocalFioDataPathEnvKey, localFioDataPathStr)
+		log.Printf("   Set %s (=%q) to the fio data path on the docker host (defaults to %v, if not running in a dev container)", HostFioDataPathEnvKey, os.Getenv(HostFioDataPathEnvKey), LocalFioDataPathEnvKey)
+
+		return nil, errors.Wrap(err, "fio setup could not be validated")
+	}
+
+	return fr, nil
+}
+
+func (fr *Runner) verifySetupWithTestWrites() error {
+	var subDirPath = path.Join("test", "subdir")
+
+	const (
+		maxTestFiles = 5
+		fileSizeB    = 1 << 20 // 1 MiB
+	)
+
+	nrFiles := rand.Intn(maxTestFiles) + 1
+
+	opt := Options{}.WithNumFiles(nrFiles).WithFileSize(fileSizeB)
+
+	err := fr.WriteFiles(subDirPath, opt)
+	if err != nil {
+		return errors.Wrap(err, "unable to perform writes")
+	}
+
+	defer fr.DeleteRelDir("test") //nolint:errcheck
```

Comment:
Would this work if `defer fr.DeleteRelDir("test")` were prior to `fr.WriteFiles`? What happens when `WriteFiles` returns an error after having written some files, but not completing successfully?

Reply:
good call, will move it